wasm/zlib/benchmark.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2025 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+class Benchmark {
+  async runIteration() {
+    if (!Module._main)
+      await setupModule(Module);
+
+    Module.FS.writeFile('input', Module.wasmBinary);
+
+    const inputStr = Module.stringToNewUTF8('input');
+    const inputzStr = Module.stringToNewUTF8('input.z');
+    const inputzoutStr = Module.stringToNewUTF8('input.z.out');
+
+    for (let i = 0; i < 10; i++) {
+      Module._compressFile(inputStr, inputzStr);
+      Module._decompressFile(inputzStr, inputzoutStr);
+      if (Module.FS.stat('input').size !== Module.FS.stat('input.z.out').size) {
+        throw new Error("Length after decompression doesn't match");
+      }
+    }
+  }
+
+  validate() {
+    const input = Module.FS.readFile('input');
+    const outputRoundtrip = Module.FS.readFile('input.z.out');
```

Comment:
Presumably these don't need to be `free`d since they just return the JS string to you but maybe worth double checking?

Reply:
I checked the Emscripten-generated code in `zlib.js` (`FS.readFile` which calls `UTF8ArrayToString`). As you say, it returns a JS string, so nothing to manually free here.